dask/dataframe/io/tests/test_sql.py

```diff
@@ -49,7 +49,7 @@ def test_npartitions(db):
     assert (data.name.compute() == df.name).all()
     data = read_sql_table('test', db, columns=['name'], npartitions=6,
                           index_col="number").compute()
-    assert (data.name == df.name).all()
+    pd.util.testing.assert_frame_equal(data, df[['name']])
```

Comment:
Typically we don't call compute, and instead do the following:

```python
from dask.dataframe.utils import assert_eq
data = read_sql_table('test', db, npartitions=6, index_col="number")   # don't call compute on this

assert_eq(data, df)
```

You can give `assert_eq` dask dataframes as well as pandas dataframes.  If you give it a dask dataframe then it will run a variety of consistency checks on it, such as that its metadata match the metadata of the computed result, that all of the keys within the dask graph conform to certain standards, etc..